src/pages/Admin/Admin_v2.tsx

```diff
@@ -34,6 +35,11 @@ const Admin_v2: React.FC = () => {
 
   const { isAdminUser } = useIsAdminUser(user?.uid);
 
+  useEffect(() => {
+    // when URL is a catch all, replace it with the canonical form
+    window.history.replaceState(null, "", adminNGRootUrl());
+  });
+
```

Comment:
NB: This code cleans up the URL in case of a catch all route hit i.e. will replace URLs like `/admin-ng/asfdsafsdafsdf` with `/admin-ng`

Reply:
`react-router` should have something like this to avoid manual logic in every single screen. I've used it before but I can't really recall the name of the component.

Reply:
I think it's basically putting a route with no path at the end under the `<Switch>` .
But from what I remember for actual replacing the history:

`const history  = useHistory();` 

was noted in some StackOverflow posts as "not always works, use native `window.history`"

Reply:
Removed in 2c33b6b8de126d7a4550fdd2ca5b20e1ad2b5168